lib/xform.js

```diff
@@ -26,6 +29,38 @@ var XFormState = State.extend(function (self, name, opts) {
         user or ``null`` if no address was chosen. If ``next`` is ``null`` or
         not defined, the state machine will be left in the current state.
         See :meth:`State.set_next_state`. Defaults to ``null``
+    :param string opts.xforms_service_url:
+        The URL where the XForms service is located. Defaults to
+        ``prd-vumi-xforms.aws.prk-host.net``.
+    :param string opts.xforms_service_error_message:
+        The error message sent back to the user if the XForms service is
+        inaccessable. Defaults to
+        ``Error contacting the xforms service``.
+    :param string opts.xform:
+        The XForm that is processed. Required.
+    :param string opts.xform_url:
+        The URL to fetch the xform from. If specified, it overrides the
+        ``opts.xform`` field. Required if ``opts.xform`` isn't specified.
+    :param string opts.xform_url_username:
+        The username to use when fetching the XForm from the XForm URL.
+        If not specified, no authentication is sent.
+    :param string opts.xform_url_password:
+        The password to use when fetching the XForm from the XForm URL.
+        If not specified, no authentication is sent.
+    :param string opts.xform_error_message:
+        The error message that is sent to the user if an xform cannot be
+        retreived from the XForm URL. Defaults to
+        ``Error in getting form from server``.
+    :param string opts.contact_namespace:
+        The namespace to user when storing the answer data onto the contact
+        under extras. If not specified, no data is saved onto the contact.
+    :param string opts.results_url:
```

Comment:
Perhaps we should pass the completed xform to the `next` function the same value we pass the `content` or `choice` to FreeText or ChoiceState `next` functions?